packages/clay-core/src/tree-view/__tests__/useTree.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * SPDX-FileCopyrightText: © 2019 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {createImmutableTree} from '../useTree';
+
+describe('useTree', () => {
+	it('patching the tree does not change the original tree', () => {
+		const tree = [
+			{
+				children: [
+					{
+						children: [
+							{
+								name: 'E',
+							},
+						],
+						name: 'C',
+					},
+					{name: 'D'},
+				],
+				name: 'A',
+			},
+			{name: 'B'},
+		];
+
+		const immutableTree = createImmutableTree(tree, 'children');
+
+		immutableTree.produce([0, 1], [0, 0]);
+
+		expect(immutableTree.applyPatches()).not.toMatchObject(tree);
+	});
+
+	it('can move items from the root of the tree', () => {
+		const tree = [
+			{
+				children: [{name: 'Blogs'}, {name: 'Documents and Media'}],
+				name: 'Liferay Drive',
+			},
+			{
+				children: [{name: 'Blogs'}, {name: 'Documents and Media'}],
+				name: 'Repositories',
+			},
+			{
+				children: [{name: 'PDF'}, {name: 'Word'}],
+				name: 'Documents and Media',
+			},
+		];
```

Comment:
This case was probably always passed because it only charges one level of nesting, maybe a test with a deeper nesting might be better.

Reply:
Sure I'll add a test for that too.

Reply:
I added a test for that in https://github.com/liferay/clay/commit/a118a2aceb31663fc1f06dc822b1754819f5973f, (I also removed one of my previous changes too), seems to work like it's supposed to.